src/python/pants/subsystem/subsystem.py

```diff
@@ -169,3 +174,55 @@ def get_options(self):
     :API: public
     """
     return self._scoped_options
+
+  @staticmethod
+  def get_streaming_workunit_callbacks(subsystem_names: Iterable[str]) -> List[Callable]:
+    """
+    This method is used to dynamically generate a list of callables
+    intended to be passed to StreamingWorkunitHandler. The caller provides a
+    collection of strings representing a Python import path to a class that
+    implements the `Subsystem` class. It will then inspect these classes for
+    the presence of a special method called `handle_workunits`, which expects a
+    single non-self argument - namely, a tuple of Python dictionaries
+    representing workunits.
+
+    For instance, you might invoke this method with something like:
+
+    `Subsystem.get_streaming_workunit_callbacks(["pants.reporting.workunits.Workunit"])`
+
+    And this will result in the method attempting to dynamically-import a
+    module called "pants.reporting.workunits", inspecting it for the presence
+    of a class called `Workunit`, getting a global instance of this Subsystem,
+    and returning a list containing a single reference to the
+    `handle_workunits` method defined on it - and returning an empty list and
+    emitting warnings if any of these steps fail.
+    """
+
+    callables = []
+
+    for name in subsystem_names:
+      try:
+        name_components = name.split(".")
+        module_name = ".".join(name_components[:-1])
+        class_name = name_components[-1]
+        module = importlib.import_module(module_name)
+        subsystem_class = getattr(module, class_name)
+      except (IndexError, AttributeError, ModuleNotFoundError, ValueError) as e:
+        logger.warning(f"Invalid module name: {name}: {e}")
+        continue
+      except ImportError as e:
+        logger.warning(f"Could not import {module_name}: {e}")
+        continue
+      try:
+        subsystem = subsystem_class.global_instance()
+      except AttributeError:
+        logger.warning(f"{subsystem_class} is not a global subsystem.")
+        continue
+
+      try:
+        callables.append(subsystem.handle_workunits)
```

Comment:
Is there some way we can define an interface or mixin or something of the sort to capture all these constraints? (i.e. the component should have a method called `handle_workunits`, it should be a global subsystem...).